cmd/experimental/kjobctl/pkg/cmd/list/list_job.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package list
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/client-go/kubernetes/scheme"
+	batchv1 "k8s.io/client-go/kubernetes/typed/batch/v1"
+	"k8s.io/utils/clock"
+
+	"sigs.k8s.io/kueue/cmd/experimental/kjobctl/pkg/cmd/util"
+	"sigs.k8s.io/kueue/cmd/experimental/kjobctl/pkg/constants"
+)
+
+const (
+	jobExample = `  # List Job
+  kjobctl list job`
+)
+
+type JobOptions struct {
+	Clock      clock.Clock
+	PrintFlags *genericclioptions.PrintFlags
+
+	Limit              int64
+	AllNamespaces      bool
+	Namespace          string
+	ProfileFilter      string
+	FieldSelector      string
+	LabelSelector      string
+	ClusterQueueFilter string
+
+	Client batchv1.BatchV1Interface
+
+	genericiooptions.IOStreams
+}
+
+func NewJobOptions(streams genericiooptions.IOStreams, clock clock.Clock) *JobOptions {
+	return &JobOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+		Clock:      clock,
+	}
+}
+
+func NewJobCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams, clock clock.Clock) *cobra.Command {
+	o := NewJobOptions(streams, clock)
+
+	cmd := &cobra.Command{
+		Use:                   "job [--selector key1=value1] [--field-selector key1=value1] [--all-namespaces]",
+		DisableFlagsInUseLine: true,
+		Short:                 "List Job",
+		Example:               jobExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter))
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addAllNamespacesFlagVar(cmd, &o.AllNamespaces)
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+
+	cmd.Flags().StringVarP(&o.ProfileFilter, "profile", "p", "",
+		"Filter by profile name which is associated with the resource.")
```

Comment:
it might be useful to put this in a helper as well, could be potentially used by `describe`

Reply:
Moved to common helper too.